iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/procedure/impl/node/RemoveDataNodesProcedure.java

```diff
@@ -123,8 +132,10 @@ protected Flow executeFromState(ConfigNodeProcedureEnv env, RemoveDataNodeState
   }
 
   private void submitChildRegionMigrate(ConfigNodeProcedureEnv env) {
-    migratedDataNodeRegions.forEach(
-        regionId -> {
+    regionMigrationPlans.forEach(
+        regionMigrationPlan -> {
+          TConsensusGroupId regionId = regionMigrationPlan.getRegionId();
+          TDataNodeLocation removedDataNode = regionMigrationPlan.getFromDataNode();
           TDataNodeLocation destDataNode =
```

Comment:
Can we abstract this transformation from removeDataNodes to RegionMigrationPlan into interface to do so that it can be connected to future optimizations? 

In the PR could write a for loop in the method to perform a ` env. GetRegionMaintainHandler ()
.filterDataNodeWithOtherRegionReplica(regionId, destDataNode)
.orElse(removedDataNode); `

Reply:
Good suggestion! This is a very important optimization, but considering the time limit, I recommend detailed design and optimization in V2.